### ISSUE TITLE
Question about finetune loss

### ISSUE THREAD

**Issue by falling-into-you** [thuml/Large-Time-Series-Model#24]
Hi. I am studying your code and I noticed that there are two kinds of losses for different inputs during finetune training. This part is easy to understand when use_ims=False.
But when use_ims=True, why do you need to calculate the loss for the entire sample? I don't quite understand what the comment means, can you provide more detailed insights?
![image](https://github.com/user-attachments/assets/846c5d35-6f2d-4489-8874-27710cb5da08)

**Comment by falling-into-you** [comment]
I further examined the code, and this difference primarily stems from the content of CIDatasetBenchmark and CIAutoRegressionDatasetBenchmark.

For CIAutoRegressionDatasetBenchmark, its seq_y contains a certain amount of historical data (label_len, 576) and prediction data (pred_len, 96), with a total length of seq_len (672).

In the inference stage, I noticed the autoregressive characteristic specific to Timer. When updating batch_x, it discards the first pred_len time steps and concatenates the last data point of pred_y. This makes the structure of batch_x essentially (a certain amount of historical data + the latest prediction result, 672).

To ensure consistency between inference and training, the same data structure is used for loss calculation during the fine-tuning process. However, it seems that the autoregressive characteristic (specifically, incorporating the prediction results back into the training process) is not utilized during fine-tuning.

Is my understanding correct? Thank you very much for your response!

**Comment by WenWeiTHU** [comment]
During training, each predicted token is supervised (including the lookback window). During inference, the code only uses the maximum context length for prediction. In fact, any input lengths less than the context length can be used for inference.
